dpe/src/commands/destroy_context.rs

```diff
@@ -227,6 +249,73 @@ mod tests {
         assert_eq!(dpe.contexts[0].children, 1 << 2);
     }
 
+    #[test]
+    fn test_retired_parent_contexts_destroyed() {
+        let mut env = DpeEnv::<TestTypes> {
+            crypto: OpensslCrypto::new(),
+            platform: DefaultPlatform,
+        };
+        let mut dpe = DpeInstance::new(&mut env, SUPPORT).unwrap();
+
+        // create new context while preserving auto-initialized context
+        let handle_1 = match (DeriveChildCmd {
+            handle: ContextHandle::default(),
+            data: [0u8; DPE_PROFILE.get_tci_size()],
+            flags: DeriveChildFlags::RETAIN_PARENT | DeriveChildFlags::CHANGE_LOCALITY,
+            tci_type: 0,
+            target_locality: TEST_LOCALITIES[1],
+        })
+        .execute(&mut dpe, &mut env, TEST_LOCALITIES[0])
+        {
+            Ok(Response::DeriveChild(resp)) => resp.handle,
+            Ok(_) => panic!("Invalid response type"),
+            Err(e) => Err(e).unwrap(),
+        };
+
+        // retire context with handle 1 and create new context
+        let handle_2 = match (DeriveChildCmd {
+            handle: handle_1,
+            data: [0u8; DPE_PROFILE.get_tci_size()],
+            flags: DeriveChildFlags::empty(),
+            tci_type: 0,
+            target_locality: TEST_LOCALITIES[1],
+        })
+        .execute(&mut dpe, &mut env, TEST_LOCALITIES[1])
+        {
+            Ok(Response::DeriveChild(resp)) => resp.handle,
+            Ok(_) => panic!("Invalid response type"),
+            Err(e) => Err(e).unwrap(),
+        };
+
+        // retire context with handle 2 and create new context
+        let handle_3 = match (DeriveChildCmd {
+            handle: handle_2,
+            data: [0u8; DPE_PROFILE.get_tci_size()],
+            flags: DeriveChildFlags::empty(),
+            tci_type: 0,
+            target_locality: TEST_LOCALITIES[1],
+        })
+        .execute(&mut dpe, &mut env, TEST_LOCALITIES[1])
+        {
+            Ok(Response::DeriveChild(resp)) => resp.handle,
+            Ok(_) => panic!("Invalid response type"),
+            Err(e) => Err(e).unwrap(),
+        };
+
+        DestroyCtxCmd { handle: handle_3 }
+            .execute(&mut dpe, &mut env, TEST_LOCALITIES[1])
+            .unwrap();
+
+        // only the auto-initialized context should remain, context[1] and context[2] should be
+        // destroyed since they are in the chain of consecutive retired parents of the destroyed
+        // context.
+        assert_eq!(
+            dpe.count_contexts(|ctx| ctx.state != ContextState::Inactive)
```

Comment:
Should this also verify that it isn't in the retired state?

Reply:
Done.